ngraph/core/src/pass/visualize_tree.cpp

```diff
@@ -328,27 +329,83 @@ static std::string pretty_partial_shape(const PartialShape& shape)
     return ss.str();
 }
 
+template <typename T>
+static std::string pretty_min_max_denormal_value(const vector<T>& values)
+{
+    std::stringstream ss;
+
+    T min_value = values[0];
+    T max_value = values[0];
+    size_t denormal_counts = 0ul;
+    std::stringstream denormal_ss;
+    for (size_t i = 0; i < values.size(); ++i)
+    {
+        const auto& value = values[i];
+        if (min_value > value)
+        {
+            min_value = value;
+        }
+        if (max_value < value)
+        {
+            max_value = value;
+        }
+
+        const auto abs_value = std::abs(static_cast<double>(value));
+        if (((abs_value > 0.) && (abs_value < 1.e-32)) || (abs_value > 1.e+32))
+        {
+            if (denormal_counts < 3)
+            {
+                denormal_ss << (denormal_counts > 0 ? ", " : "") << i << ": " << value;
```

Comment:
Why do you add only first 3 denormal elements?
What if vector contains not only denormals?

Reply:
> Why do you add only first 3 denormal elements?  

I don't have any requests from other team members, first 3 elements are enough for me : )
> What if vector contains not only denormals?  

Original values are not affected. Denormal values will be displayed at the end of the original box.